singularity-eos/eos/eos_builder.hpp

```diff
@@ -15,129 +15,57 @@
-  modifiers_t modifiers;
-  return buildEOS(type, base_params, modifiers);
-}
-
-template <typename T>
-EOS makeUnitSystem(T &&eos, bool use_length_time, Real rho_unit, Real sie_unit,
-                   Real temp_unit, Real time_unit, Real mass_unit, Real length_unit) {
-  if (use_length_time) {
-    return UnitSystem<T>(std::move(eos), eos_units_init::length_time_units_init_tag,
-                         time_unit, mass_unit, length_unit, temp_unit);
-  } else {
-    return UnitSystem<T>(std::move(eos), eos_units_init::thermal_units_init_tag, rho_unit,
-                         sie_unit, temp_unit);
-  }
-}
+// Just import the variadic utils namespace int the EOSBuilder
+// namespace.
+using namespace variadic_utils;
 
-template <typename T>
-EOS makeRelativistic(T &&eos, Real cl) {
-  return RelativisticEOS<T>(std::move(eos), cl);
+template <template <class...> typename Mod, typename U, typename... Ts>
+using IsModifiable =
+    conjunction<contains<U, Ts...>::value, contains<Mod<U>, Ts...>::value>;
+template <template <class...> typename Mod, typename U, typename... Ts>
+constexpr bool is_modifiable(const type_list<Ts...> &var) {
+  return IsModifiable<Mod, U, Ts...>::value;
 }
-
-template <typename T>
-EOS makeBilinearRamp(T &&eos, Real r0, Real a, Real b, Real c) {
-  return BilinearRampEOS<T>(std::forward<T>(eos), r0, a, b, c);
+template <template <class...> typename Mod, typename U, typename... Ts>
+constexpr bool is_modifiable(const Variant<Ts...> &var) {
+  return IsModifiable<Mod, U, Ts...>::value;
 }
-
-template <typename T>
-EOS applyShiftAndScale(T &&eos, bool scaled, bool shifted, Real scale, Real shift) {
-  if (shifted && scaled) {
-    ShiftedEOS<T> a(std::forward<T>(eos), shift);
-    ScaledEOS<ShiftedEOS<T>> b(std::move(a), scale);
-    return b;
-  }
-  if (shifted) {
-    return ShiftedEOS<T>(std::forward<T>(eos), shift);
-  }
-  if (scaled) {
-    return ScaledEOS<T>(std::forward<T>(eos), scale);
-  }
-  return eos;
+template <template <class...> typename Mod, typename U, typename... Ts>
+constexpr bool is_modifiable(const U &u, const Variant<Ts...> &var) {
+  return IsModifiable<Mod, U, Ts...>::value;
 }
 
-template <typename T, template <typename> class W, typename... ARGS>
-EOS applyWrappedShiftAndScale(T &&eos, bool scaled, bool shifted, Real scale, Real shift,
-                              ARGS... args) {
-  if (shifted && scaled) {
-    ShiftedEOS<T> a(std::forward<T>(eos), shift);
-    ScaledEOS<ShiftedEOS<T>> b(std::move(a), scale);
-    W<ScaledEOS<ShiftedEOS<T>>> c(std::move(b), args...);
-    return c;
-  }
-  if (shifted) {
-    ShiftedEOS<T> sh_eos(std::forward<T>(eos), shift);
-    return W<ShiftedEOS<T>>(std::move(sh_eos), args...);
-  }
-  if (scaled) {
-    ScaledEOS<T> sc_eos(std::forward<T>(eos), scale);
-    return W<ScaledEOS<T>>(std::move(sc_eos), args...);
-  }
-  return W<T>(std::forward<T>(eos), args...);
+// Modifies the eos contained in the var object with the modifier Mod,
```

Comment:
what is going on with this commented code? The git diff makes it a bit confusing for me to figure out.

Reply:
That's example intended usage. This is documented in the docs now. I'll just remove the comment.